### PR TITLE
[UI]: update wording on STAX

### PR DESCRIPTION
**Pull request by libotony**
requires #18

### CODE REVIEW

File: src/ui/ui_nbgl.c [hunk 1]
```diff
@@ -172,40 +172,61 @@ void ui_display_public_key_flow(void) {
...
     else 
     {
-        nbgl_useCaseStatus("Transaction rejected", false, ui_idle);
+        if(sign_type == SIGN_TRANSACTION){
+            nbgl_useCaseStatus("Transaction rejected", false, ui_idle);
+        }else if(sign_type == SIGN_MESSAGE){
+            nbgl_useCaseStatus("Message rejected", false, ui_idle);
+        }else{
+            nbgl_useCaseStatus("Certificate rejected", false, ui_idle);
+        }
```
Comment (tdejoigny-ledger) [inline src/ui/ui_nbgl.c:200]: As discussed here you should be aligned with the design guideline and keep the generic wording "transaction signed" or "transacton rejected"

File: src/ui/ui_nbgl.c [hunk 2]
```diff
@@ -172,40 +172,61 @@ void ui_display_public_key_flow(void) {
...
 static void reject_confirmation(void) {
-    nbgl_useCaseConfirm("Reject transaction?", NULL, "Yes, Reject", "Go back to transaction", transaction_rejected);
+    if(sign_type == SIGN_TRANSACTION){
+        nbgl_useCaseConfirm("Reject transaction?", NULL, "Yes, Reject", "Go back to transaction", sign_rejected);
+    }else if(sign_type == SIGN_MESSAGE){
+        nbgl_useCaseConfirm("Reject message?", NULL, "Yes, Reject", "Go back to message", sign_rejected);
+    }else{
+        nbgl_useCaseConfirm("Reject certificate?", NULL, "Yes, Reject", "Go back to certificate", sign_rejected);
+    }
 }
```
Comment (tdejoigny-ledger) [inline src/ui/ui_nbgl.c:218]: same comment here.

File: src/ui/ui_nbgl.c [hunk 3]
```diff
@@ -319,51 +338,61 @@ static void review_msg_certif_choice(bool confirm) {
...
-    transaction_type = p_transaction_type;
-    if(transaction_type == MSG_TRANSACTION)
+    sign_type = p_sign_type;
+    if(sign_type == SIGN_MESSAGE)
     {
-        transaction_type_to_display = "Review message";
+        nbgl_useCaseReviewStart(&C_Message_64px,
+                            "Review message",
+                            NULL,
+                            "Reject message",
```
Comment (tdejoigny-ledger) [inline src/ui/ui_nbgl.c:383]: should be replaced by "reject transaction"

File: src/ui/ui_nbgl.c [hunk 4]
```diff
@@ -319,51 +338,61 @@ static void review_msg_certif_choice(bool confirm) {
...
-    }
-
-    nbgl_useCaseReviewStart(&C_Message_64px,
-                            transaction_type_to_display,
+        nbgl_useCaseReviewStart(&C_Message_64px,
+                            "Review certificate",
                             NULL,
-                            "Reject transaction",
-                            single_action_msg_certif_review,
+                            "Reject certificate",
```
Comment (tdejoigny-ledger) [inline src/ui/ui_nbgl.c:392]: should be replaced by "reject transaction"

File: src/ui/ui_nbgl.c [hunk 5]
```diff
@@ -319,51 +338,61 @@ static void review_msg_certif_choice(bool confirm) {
...
     // Info long press
-    msg_certif_info_long_press.icon = &C_Message_64px;
-    if(transaction_type == MSG_TRANSACTION)
+    msg_cert_info_long_press.icon = &C_Message_64px;
+    msg_cert_info_long_press.longPressText = "Hold to sign";
+    if(sign_type == SIGN_MESSAGE)
     {
-        msg_certif_info_long_press.text = "Sign message?";
+        msg_cert_info_long_press.text = "Sign message?";
+        nbgl_useCaseStaticReview(&msg_cert_pair_list, &msg_cert_info_long_press, "Reject message", review_msg_cert_choice);
```
Comment (tdejoigny-ledger) [inline src/ui/ui_nbgl.c:365]: "reject message" should be replaced by "reject transaction".
The bottom button should be always "reject transaction" (generic wording)

Ok for the wording change of `msg_cert_info_long_press.text`

File: src/ui/ui_nbgl.c [hunk 6]
```diff
@@ -319,51 +338,61 @@ static void review_msg_certif_choice(bool confirm) {
...
     }
     else
     {
-        msg_certif_info_long_press.text = "Sign certificate?";
+        msg_cert_info_long_press.text = "Sign certificate?";
+        nbgl_useCaseStaticReview(&msg_cert_pair_list, &msg_cert_info_long_press, "Reject certificate", review_msg_cert_choice);
```
Comment (tdejoigny-ledger) [inline src/ui/ui_nbgl.c:370]: "reject certificate" should be replaced by "reject transaction".

File: src/ui/ui_nbgl.c (hunk 1 above)
Comment (libotony) [reply src/ui/ui_nbgl.c:200]: Understood, thank you!

File: src/ui/ui_nbgl.c (hunk 2 above)
Comment (libotony) [reply src/ui/ui_nbgl.c:218]: How about the go-back hint, `Go back to transaction` or `Go back to message` ?

Comment (tdejoigny-ledger) [reply src/ui/ui_nbgl.c:218]: same for this button, should be also `transaction`

**Comment by libotony** [comment]
@tdejoigny-ledger Updated based on your comments, please take a look

**Comment by tdejoigny-ledger** [comment]
yes it is better now. I think there is an issue with the test `test_sign_certificate_cancel`, you should call `sign_certificate` instead of `sign_message`

**Comment by libotony** [comment]
> yes it is better now. I think there is an issue with the test `test_sign_certificate_cancel`, you should call `sign_certificate` instead of `sign_message`

I have another PR for this

**Pull request merged**